src/herbivoresAndCarnivores.js

```diff
@@ -1,17 +1,48 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    this.hidden = false;
+    Animal.alive.push(this);
+  }
+
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super(name);
+    Animal.alive.push(this);
+  }
+
+  bite(animal) {
+    if (animal instanceof Herbivore && animal.hidden === false) {
+      animal.health -= 50;
+
+      if (animal.health === 0) {
+        const index = Animal.alive
+          .findIndex(someAnimal => someAnimal === animal);
+
+        if (index >= 0) {
+          Animal.alive.splice(index, 1);
+        }
+      }
+    }
```

Comment:
3 nested if blocks😵

Let's make it simpler =)

Also better to avoid objects comparing, let's just filter array and remove animals with health less or equal 0 (better to check this condition instead of just equality, 'cause we can change animal health or carnivore damage and it shouldn't require any refactor)